File: WoB_Client/Assets/Network/Response/ResponseUpdateTime.cs
using UnityEngine;

using System;

public class ResponseUpdateTimeEventArgs : ExtendedEventArgs {
	public short month { get; set; }
	public int duration { get; set; }
	public int current { get; set; }
	public float rate { get; set; }
	
	public ResponseUpdateTimeEventArgs() {
		event_id = Constants.SMSG_UPDATE_TIME;
	}
}

public class ResponseUpdateTime : NetworkResponse {
	
	private short month;
	private int duration;
	private int current;
	private float rate;
	
	public ResponseUpdateTime() {
	}
	
	public override void parse() {
		month = DataReader.ReadShort(dataStream);
		duration = DataReader.ReadInt(dataStream);
		current = DataReader.ReadInt(dataStream);
		rate = DataReader.ReadFloat(dataStream);
	}
	
	public override ExtendedEventArgs process() {
		ResponseUpdateTimeEventArgs args = new ResponseUpdateTimeEventArgs();
		args.month = month;
		args.duration = duration;
		args.current = current;
		args.rate = rate;
		
		return args;
	}
}

File: WoB_Server/src/networking/response/ResponseUpdateTime.java
package networking.response;

// Other Imports
import metadata.Constants;
import utility.GamePacket;

public class ResponseUpdateTime extends GameResponse {

    private int current;
    private int duration;
    private int month;
    private float rate;

    public ResponseUpdateTime() {
        responseCode = Constants.SMSG_UPDATE_TIME;
    }

    @Override
    public byte[] constructResponseInBytes() {
        GamePacket packet = new GamePacket(responseCode);
        packet.addShort16((short) month);
        packet.addInt32(duration);
        packet.addInt32(current);
        packet.addFloat(rate);
        return packet.getBytes();
    }

    public void setCurrent(int current) {
        this.current = current;
    }

    public void setDuration(int duration) {
        this.duration = duration;
    }

    public void setMonth(int month) {
        this.month = month;
    }

    public void setRate(float rate) {
        this.rate = rate;
    }
}
